File: ts/src/http-error-format.ts
export const HTTP_ERROR_FORMAT_NESTED = "nested";
export const HTTP_ERROR_FORMAT_FLAT_LEGACY = "flat_legacy";

export type HTTPErrorFormat =
  | typeof HTTP_ERROR_FORMAT_NESTED
  | typeof HTTP_ERROR_FORMAT_FLAT_LEGACY;

export function normalizeHTTPErrorFormat(format: unknown): HTTPErrorFormat {
  return format === HTTP_ERROR_FORMAT_FLAT_LEGACY
    ? HTTP_ERROR_FORMAT_FLAT_LEGACY
    : HTTP_ERROR_FORMAT_NESTED;
}

File: runtime/http_error_format.go
package apptheory

// HTTPErrorFormat controls how AppTheory serializes HTTP error bodies.
type HTTPErrorFormat string

const (
	HTTPErrorFormatNested     HTTPErrorFormat = "nested"
	HTTPErrorFormatFlatLegacy HTTPErrorFormat = "flat_legacy"
)

func normalizeHTTPErrorFormat(format HTTPErrorFormat) HTTPErrorFormat {
	switch format {
	case HTTPErrorFormatFlatLegacy:
		return HTTPErrorFormatFlatLegacy
	default:
		return HTTPErrorFormatNested
	}
}
